pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/design/LawOfDemeterRule.java

```diff
@@ -166,19 +165,34 @@ private static boolean isNotLiteral(ASTPrimaryExpression expression) {
             return true;
         }
 
-        private boolean isNotBuilder() {
+        boolean isNotBuilder() {
             return baseType != StringBuffer.class && baseType != StringBuilder.class
                     && !"StringBuilder".equals(baseTypeName) && !"StringBuffer".equals(baseTypeName)
                     && !methodName.endsWith("Builder");
         }
 
-        private static List<ASTPrimarySuffix> findSuffixesWithoutArguments(ASTPrimaryExpression expr) {
-            List<ASTPrimarySuffix> result = new ArrayList<>();
+
+        /**
+         * Returns the children suffixes that is not arguments.
+         *
+         * @param expr expression that is be judged.
+         
+         * @return the suffix which is the child of expression without Argument.
+         */
+        private static List<ASTPrimarySuffix> findSuffixesWithoutArguments(final ASTPrimaryExpression expr) {
+            final List<ASTPrimarySuffix> result = new ArrayList<>();
             if (hasRealPrefix(expr)) {
-                List<ASTPrimarySuffix> suffixes = expr.findDescendantsOfType(ASTPrimarySuffix.class);
-                for (ASTPrimarySuffix suffix : suffixes) {
-                    if (!suffix.isArguments()) {
-                        result.add(suffix);
+                final List<ASTPrimarySuffix> suffixes = new ArrayList<>();
+                Node child;
+                for (int i = 0; i < expr.getNumChildren(); i++) {
+                    child = expr.getChild(i);
+                    if (child instanceof ASTPrimarySuffix) {
+                        suffixes.add((ASTPrimarySuffix) child);
+                    }
+                }
+                for (final ASTPrimarySuffix suffixx : suffixes) {
```

Comment:
This looks to me exactly the same as before - you only change the variable name from `suffix` to `suffixx`. If it is the same, we should not change it, since it is irrelevant then... please change it back.